tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -154,7 +154,9 @@ sealed trait NativeConfig {
   def withASAN(value: Boolean): NativeConfig
 
   /** Create a new config with given behavior for stubs. */
-  def withLinkStubs(value: Boolean): NativeConfig
+  @deprecated(
+    "This option has been deprecated and no longer have effects on the build"
+  ) def withLinkStubs(value: Boolean): NativeConfig
```

Comment:
I think we've discussed somewhere that it should remain in it's current form, without deprecation, but we should remove all unnecesery usages of `@stub`

Reply:
That also means we should restore previous behaviour from linker.Reach and tests and scaladoc comments for annotation

Reply:
I somehow thought the issue is about making `withLinkStubs` no-op, but it seems to be extended goal. Let's just focus on removing the abuse of `@stub` annotation in this PR as mentioned in https://github.com/scala-native/scala-native/issues/3214#issuecomment-1464336249